core/keyserver/keyserver.go

```diff
@@ -636,36 +635,30 @@ func (s *Server) BatchQueueUserUpdate(ctx context.Context, in *pb.BatchQueueUser
 	// - Correct key formats.
 	_, tdone := monitoring.StartSpan(ctx, "BatchQueueUserUpdate.verify")
 	updates := make(chan *pb.EntryUpdate)
-	errors := make(chan error)
 	go func() {
 		defer close(updates)
 		for _, u := range in.Updates {
 			updates <- u
 		}
 	}()
-	go func() {
-		defer close(errors)
-		var wg sync.WaitGroup
-		defer wg.Wait() // Wait before closing errors
-		for w := 0; w < runtime.NumCPU(); w++ {
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-				for u := range updates {
-					if err := s.verifyMutation(u.Mutation); err != nil {
-						glog.Warningf("Invalid UpdateEntryRequest: %v", err)
-						errors <- status.Errorf(codes.InvalidArgument, "Invalid mutation")
-					}
-					if err = validateEntryUpdate(u, vrfPriv); err != nil {
-						glog.Warningf("Invalid UpdateEntryRequest: %v", err)
-						errors <- status.Errorf(codes.InvalidArgument, "Invalid request")
-					}
-				}
-			}()
-		}
-	}()
-	for e := range errors {
-		return nil, e
+	g, _ := errgroup.WithContext(ctx)
+	for u := range updates {
```

Comment:
Btw, the `updates` channel seems redundant here, this could iterate the `in.Updates` slice directly.

Reply:
good point